lib/fluent/plugin_helper/socket_option.rb

```diff
@@ -49,9 +51,21 @@ def socket_option_set(sock, resolve_name: nil, nonblock: false, linger_timeout:
         if nonblock
           sock.fcntl(Fcntl::F_SETFL, Fcntl::O_NONBLOCK)
         end
-        if linger_timeout
+        if Fluent.windows?
+          # To prevent closing socket forcibly on Windows,
+          # this options shouldn't be set up when linger_timeout equals to 0 (including nil).
+          # This unintended behavior always ocurrs on Windows when linger_timeout.to_i == 0.
+          # This unintented behavior causes "Errno::ECONNRESET: An existing connection was forcibly
+          # closed by the remote host." on Windows.
+          if linger_timeout.to_i > 0
+            optval = [1, linger_timeout.to_i].pack(FORMAT_STRUCT_LINGER_WINDOWS)
```

Comment:
It may be better to check `linger_timeout` should be less than 2^16? or this case won't happen?

Reply:
Hmm..., specifying `2**17` here does not cause `Errno::ECONNRESET`.
But it maybe reasonable to check greater than 65535(2**16).

Reply:
Fixed in https://github.com/fluent/fluentd/pull/2862/commits/b55e610909943cce71dda04c242a87f573fc1be9